tools/fast/series/sleep_delay_from_ctx.go

```diff
@@ -0,0 +1,34 @@
+package series
+
+import (
+	"context"
+	"time"
+
+	"github.com/filecoin-project/go-filecoin/mining"
+)
+
+var (
+	// Key used to set the time.Duration in the context
+	sleepDelayKey = struct{}{}
+
+	// Default delay
+	defaultSleepDelay = mining.DefaultBlockTime
+)
+
+// SetSleepDelay returns a context with `d` set in the context. To sleep with
+// the value, call SleepDelayFromCtx with the context.
+func SetSleepDelay(ctx context.Context, d time.Duration) context.Context {
+	return context.WithValue(ctx, sleepDelayKey, d)
+}
+
+// SleepDelay is a helper method to make sure people don't call `time.Sleep`
```

Comment:
Should be "SleepDelayFromCtx" to match function name (a bit concerned this passed limiting).

The naming here leads me to believe `SleepDelayFromCtx` will return a `duration` from the context, could we either:
1. Rename this to indicate calling the method will perform the sleeping: `SleepForCtxDelay` or similar
2. Return the duration from the context and leave the method naming as it. A very similar pattern is [used here](https://github.com/opentracing/opentracing-go/blob/master/gocontext.go#L5-L27). My preference is this option with the method explicitly wrapped in a sleep when called.

Reply:
What about `CtxSleep`?

Reply:
Either sound good to me (rename or wrap in sleep call), I will leave this decision up to you.